<launch>
  <arg name="name" default="quadrotor"/>
  <arg name="type" default="hummingbird"/>
  <arg name="mass" default="0.5"/>
  <arg name="world_frame_id" default="simulator"/>
  <arg name="rate/odom" default="500"/>
  <arg name="color/r" default="0.0"/>
  <arg name="color/g" default="0.0"/>
  <arg name="color/b" default="1.0"/>
  <arg name="color/a" default="0.7"/>

  <node pkg="flightmare_simulator"
    type="quadrotor_simulator_so3"
    name="quadrotor_simulator_so3"
    output="screen">
    <rosparam file="$(find flightmare_simulator)/config/$(arg type)_params.yaml"/>
    <rosparam file="$(find flightmare_simulator)/config/aprilTag_config_detect.yaml"/>
    <param name="world_frame_id" value="$(arg world_frame_id)"/>
    <param name="quadrotor_name" value="$(arg name)"/>
    <param name="rate/odom" value="$(arg rate/odom)"/>
    <param name="mass" value="$(arg mass)"/>
    <remap from="~odom" to="odom"/>
    <remap from="~imu" to="imu"/>
    <remap from="~cmd" to="so3_cmd"/>
  </node>

  <node pkg="mesh_visualization"
    type="mesh_visualization"
    name="mesh_visualization"
    output="screen">
    <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
    <param name="color/r" value="$(arg color/r)"/>
    <param name="color/g" value="$(arg color/g)"/>
    <param name="color/b" value="$(arg color/b)"/>
    <param name="color/a" value="$(arg color/a)"/>
    <remap from="~input" to="odom"/>
  </node>
  
    <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <remap from="image_rect" to="/quadrotor/quadrotor_simulator_so3/unity_drone_cam" />
    <remap from="camera_info" to="/quadrotor/quadrotor_simulator_so3/unity_drone_cam/camera_info" />
    <param name="image_transport" type="str" value="compressed" />
    <param name="tag_family" type="str" value="36h11" />
    <param name="projected_optics" type="bool" value="true" />
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.4},
      {id: 1, size: 0.4},
      {id: 2, size: 0.4},
      {id: 3, size: 0.4},
      {id: 4, size: 0.4},
      {id: 5, size: 0.4}
]
    </rosparam>
  </node>
</launch>
